src/twister2/device/device_abstract.py

```diff
@@ -28,23 +35,40 @@ def env(self) -> dict[str, str]:
         return env
 
     @abc.abstractmethod
-    def connect(self):
+    def connect(self) -> None:
         pass
 
     @abc.abstractmethod
-    def disconnect(self):
+    def disconnect(self) -> None:
         pass
 
-    @abc.abstractmethod
     def flash(self, build_dir: str | Path, timeout: float = 60.0) -> None:
         """
         Flash and run code on a device.
 
         :param build_dir: build directory
         :param timeout: time out in seconds
         """
+        logger.info('Flashing device')
+        self.build_dir = build_dir
+        self.timeout = timeout
+        self.start()
 
     @property
     @abc.abstractmethod
     def out(self) -> Generator[str, None, None]:
         """Return output from a device."""
+
+    def join(self, timeout: float = None) -> None:
+        super().join(timeout=1)
```

Comment:
Added comment in the code (which was also modified).
This timeout is used to avoid situation that for some reason tread got stuck (we should not wait forever here).